### PR TITLE
CMake build and test.

### PR DESCRIPTION
**Pull request by spencels**
Add support for building and testing with CMake. This is necessary for future MSVC support (#300).

Not supported yet:
* Windows
* Emscripten
* Python
* Compilers other than GCC and Clang.

### CODE REVIEW

File: core/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,67 @@
+# libjsonnet
+
+# Remember to update Bazel and Makefile builds when updating this list!
```
Comment (sparkprime) [inline core/CMakeLists.txt]: and MANIFEST.in and setup.py :(

**Comment by sparkprime** [comment]
This looks ok, except it seems your build test is failing on mac?

I'd like to try it out myself, but haven't got time right now.

In particular, is there enough feature parity to remove the handwritten makefiles...

**Comment by sparkprime** [comment]
Oh there is emscripten, as you said.

**Comment by thefoojoo** [comment]
CMake could easily take over everything the handwritten makefile does now. The Python build be a CMake target, where CMake [generates setup.py/MANIFEST.in via a template](https://bloerg.net/2012/11/10/cmake-and-distutils.html) using the LIB_SOURCES variable in `core/CMakeLists.txt`. I believe there are CMake libraries out there for finding Python installations. I'm pretty sure CMake could also do the Emscripten build as well, but I'm less certain on how that would work. 

I'd prefer to implement Python and Emscripten builds in a follow-up PR to keep the size manageable, but I could build it into this one if you'd like.

I think Travis is failing on Linux. I developed this on a Mac so at least the Mac build is green :). I have access to all three systems so I will troubleshoot it when I get a chance.

**Comment by thefoojoo** [comment]
I fixed the issue with Linux, Travis should work now (I'm not sure why it hasn't re-run). I also added a cross-platform std.jsonnet.h generator.

I noticed some functionality in run_tests.sh. I think I can implement that in CMake. Shell scripts won't run on Windows so I'll want to port that eventually.

**Comment by sparkprime** [comment]
There is one script that makes sure jsonnet outputs "true" (or matches the golden file if provided) and another script that makes sure jsonnet fmt does not change the jsonnet file (or matches the golden file if provided).

**Comment by sparkprime** [comment]
Hey, should I have merged this?

**Comment by spencels** [comment]
Yes, I think this was ready to go out. I'll resolve merge conflicts and give it a quick test in the next week and let you know when it's ready.

**Comment by sparkprime** [comment]
Awesome, thanks for coming back to it after all this time.

**Comment by spencels** [comment]
The code looks good overall, but tests are failing after the pulling in 8 months-worth of changes. I'll ping this thread when I have it working again.

**Comment by spencels** [comment]
The error was on my end. This is ready to go.

**Comment by sparkprime** [comment]
Thanks!

**Pull request merged**